bolts/data/datasets/audio/ecoacoustics.py

```diff
@@ -1 +1,224 @@
+            paths: list[Path] = []
+            paths.extend(self.base_dir.glob(f"**/*{ext}"))
+            str_paths = pd.Series([str(path.relative_to(self.base_dir)) for path in paths])
+
+            df = str_paths.str.rpartition(
+                "\\",
+            )
+            df[0] = df[0] + df[1]
+            return df.drop(columns=[1]).rename(columns={0: "filePath", 2: "fileName"})
+
+        ec_labels = pd.read_csv(self.ec_labels_path, encoding="ISO-8859-1")
+        uk_labels = pd.read_csv(self.uk_labels_path, encoding="ISO-8859-1")
+
+        sgram_seg_metadata = gen_files_df(".pt")
+        sgram_seg_metadata.sort_values(by=["fileName"], axis=0, inplace=True)
+
+        # Merge labels and metadata files.
+        metadata = gen_files_df(".wav")
+        metadata = metadata.merge(pd.concat([uk_labels, ec_labels]), how="left")
+        metadata = metadata.loc[metadata.index.repeat(self._n_sgram_segments)]
+        metadata.sort_values(by=["fileName"], axis=0, inplace=True)
+
+        # Replace metadata filename and filepath column values with those in sgram_seg_info.
+        metadata["filePath"] = sgram_seg_metadata["filePath"].values
+        metadata["fileName"] = sgram_seg_metadata["fileName"].values
+
+        metadata.reset_index(inplace=True, drop=True)
+        metadata = self._label_encode_metadata(metadata)
+        metadata.to_csv(self._metadata_path)
+
+    def _preprocess_audio(
+        self,
+        specgram_segment_len: float,
+        new_sr: int,
+        n_freq_bins: int,
+        hop_len: int,
+        transform: Callable[[Tensor], Tensor],
+    ) -> None:
+        """
+        Applies transformation to audio samples then segments transformed samples and stores
+        them as processed files.
+        """
+
+        if not self._processed_audio_dir.exists():
+            mkdir(self._processed_audio_dir)
+
+        waveform_paths = list(self.base_dir.glob("**/*.wav"))
+
+        tform = transform(n_fft=n_freq_bins, hop_length=hop_len)
+        for path in waveform_paths:
+            waveform_filename = str(path).split("\\")[-1].split(".")[0]
+            waveform, sr = torchaudio.load(path)
+            waveform = F.resample(waveform, sr, new_sr)
+            specgram = tform(waveform)
+
+            spectrogram_segments = self._segment_spectrogram(specgram, specgram_segment_len)
+            for i, segment in enumerate(spectrogram_segments):
+                torch.save(segment, f"{str(self._processed_audio_dir)}\\{waveform_filename}_{i}.pt")
+
+    def _segment_spectrogram(self, specgram: Tensor, segment_len: float) -> list[Tensor]:
+        """
+        Takes a spectrogram and segments it into as many segments of segment_len width as possible.
+        """
+        seg_sz = int(specgram.shape[-1] / (self._AUDIO_LEN / segment_len))
+        segment_boundaries = [(i - seg_sz, i) for i in range(seg_sz, specgram.shape[-1], seg_sz)]
+        return [specgram[:, :, start:end] for start, end in segment_boundaries]
```

Comment:
```suggestion
        segment_boundaries = [(i - seg_sz, i) for i in range(seg_sz, specgram.shape[-1], seg_sz)]
        return [specgram[:, :, start:end] for start, end in segment_boundaries]
```
return specgram.split(split_size_or_sections=seg_sz, dim=-1)